src/SonarScanner.Shim/PropertiesWriter.cs

```diff
@@ -284,6 +284,27 @@ private static bool IsAscii(char c)
             return c <= sbyte.MaxValue;
         }
 
+        internal /* for testing purposes */ string JoinAsSonarQubeMultiValueProperty(IEnumerable<string> paths)
+        {
+            if (Version.TryParse(this.config.SonarQubeVersion, out var sonarqubeVersion) &&
+                sonarqubeVersion.CompareTo(new Version(6, 5)) >= 0)
+            {
+                return string.Join(MultiValuesPropertySeparator, paths.Select(path => $"\"{path.Replace("\"", "\"\"")}\""));
+            }
+            else
+            {
+                Func<string, bool> invalidPathPredicate = path => path.Contains(",");
+                var invalidPaths = paths.Where(invalidPathPredicate);
+                if (invalidPaths.Any())
+                {
+                    this.logger.LogWarning(Resources.WARN_InvalidCharacterInPaths, string.Join(", ", invalidPaths));
+                    return string.Join(MultiValuesPropertySeparator, paths.Where(path => !invalidPathPredicate(path)));
```

Comment:
Could have this expression as the only _return_ statement in line 304.

Reply:
I wanted to avoid the `Where()` clause but I guess it doesn't cost much.